server/src/main/java/org/opensearch/action/admin/cluster/wlm/QueryGroupStatsResponse.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.wlm;
+
+import org.opensearch.action.FailedNodeException;
+import org.opensearch.action.support.nodes.BaseNodesResponse;
+import org.opensearch.cluster.ClusterName;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.ToXContentFragment;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.wlm.stats.QueryGroupStats;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * A response for obtaining QueryGroupStats
+ */
+@ExperimentalApi
+public class QueryGroupStatsResponse extends BaseNodesResponse<QueryGroupStats> implements ToXContentFragment {
+
+    QueryGroupStatsResponse(StreamInput in) throws IOException {
+        super(in);
+    }
+
+    QueryGroupStatsResponse(ClusterName clusterName, List<QueryGroupStats> nodes, List<FailedNodeException> failures) {
+        super(clusterName, nodes, failures);
+    }
+
+    @Override
+    protected List<QueryGroupStats> readNodesFrom(StreamInput in) throws IOException {
```

Comment:
Will this send the node_id along with the `QueryGroupStats`

Reply:
This only returns List<QueryGroupStats> so no node_id

Reply:
This returns List<QueryGroupStats> but we can use queryGroupStats.getNode().getId() to get node_id